addons/external_intercom/functions/fnc_connect.sqf

```diff
@@ -0,0 +1,148 @@
+#include "script_component.hpp"
+
+/*
+  Name: TFAR_external_intercom_fnc_connect
+
+  Author: Arend(Saborknight)
+    Connects the player to the intercom of a vehicle, if intercom is enabled on
+    the vehicle.
+
+  Arguments:
+    0: Vehicle object <OBJECT>
+    1: Player unit <OBJECT>
+    2: Is Wireless connection? <BOOLEAN>
+
+  Return Value:
+    None
+
+  Example:
+    [_vehicle, _player] call TFAR_external_intercom_fnc_connect;
+
+  Public: Yes
+*/
+
+params ["_vehicle", "_player", "_actionParams"];
+_actionParams params [["_wireless", false, [false]]];
+
+if (_wireless && !((headgear _player) in TFAR_externalIntercomWirelessHeadgear)) exitWith {
+    [parseText format[CSTRING(HEADGEAR_NOT_WIRELESS_CAPABLE), getText(configFile >> "CfgWeapons" >> headgear _player >> "displayName")]] call TFAR_fnc_showHint;
+    "no";
+};
+
+private _externalSpeakers = _vehicle getVariable ["TFAR_ExternalIntercomSpeakers", [objNull, []]];
+// External Speakers structure is always:
+//  [_playerUsingPhone, [_infinitePlayersConnectedWirelessly...]]
+if (!isNull (_externalSpeakers select 0) && !_wireless) exitWith {
+    [parseText format[CSTRING(SOMEONE_USING_PHONE), name (_externalSpeakers select 0)]] call TFAR_fnc_showHint;
+    "no";
+};
+
+// Make sure we disconnect properly from any other connected vehicles
+private _oldVehicle = _player getVariable "TFAR_ExternalIntercomVehicle";
+if (!isNil {_oldVehicle}) then {
+    [_oldVehicle, _player] call TFAR_external_intercom_fnc_disconnect;
+};
+
+private _vehicleID = _vehicle getVariable ["TFAR_vehicleIDOverride", netId _vehicle];
+_vehicle setVariable ["TFAR_vehicleIDOverride", _vehicleID, true];
+_player setVariable ["TFAR_ExternalIntercomVehicle", _vehicle, true];
+_player setVariable ["TFAR_vehicleIDOverride", _vehicleID, true];
+
+if (_wireless) then {
+    (_externalSpeakers select 1) pushBack _player;
+} else {
+    _externalSpeakers set [0, _player];
+};
+_vehicle setVariable ["TFAR_ExternalIntercomSpeakers", _externalSpeakers, true];
+
+// Get intercom channel, now that we've signed on to the intercom
+private _intercomSlot = [_vehicle] call TFAR_external_intercom_fnc_getIntercomChannel;
+
+_vehicle setVariable [format ["TFAR_IntercomSlot_%1",(netId _player)], _intercomSlot, true];
+_player setVariable [format ["TFAR_IntercomSlot_%1",(netId _player)], _intercomSlot, true]; // Because TFAR checks the "vehicle", and thinks the unit is the vehicle when it's not in a vehicle, so we manually set it
+
+// Show indicator
+if (TFAR_oldVolumeHint) then {
```

Comment:
You don't need to use this. This was only specifically for Volume indicator (as some people didn't want the new on-screen indicator).

This is a new feature, you don't need to worry about backwards compat and people "preferring the old way"
I'd say take that out and only ever use the new indicator.

Or if you want to keep it, its unrelated to volume display so it should probably be separate setting then

Reply:
Oh hell no, I prefer the icon interface by far